packages/launchpad/src/encoding.spec.ts

```diff
@@ -1 +1,55 @@
-describe("encoding", () => {});
+import { sortJson } from "./encoding";
+
+describe("encoding", () => {
+  describe("sortJson", () => {
+    it("leaves non-objects unchanged", () => {
+      expect(sortJson(true)).toEqual(true);
+      expect(sortJson(false)).toEqual(false);
+      expect(sortJson("aabbccdd")).toEqual("aabbccdd");
+      expect(sortJson(75)).toEqual(75);
+      expect(sortJson(null)).toEqual(null);
+      expect(sortJson([5, 6, 7, 1])).toEqual([5, 6, 7, 1]);
+      expect(sortJson([5, ["a", "b"], true, null, 1])).toEqual([5, ["a", "b"], true, null, 1]);
+    });
+
+    it("sorts objects by key", () => {
+      // already sorted
+      expect(sortJson({})).toEqual({});
+      expect(sortJson({ a: 3 })).toEqual({ a: 3 });
+      expect(sortJson({ a: 3, b: 2, c: 1 })).toEqual({ a: 3, b: 2, c: 1 });
+
+      // not yet sorted
+      expect(sortJson({ b: 2, a: 3, c: 1 })).toEqual({ a: 3, b: 2, c: 1 });
```

Comment:
`toEqual` doesn't test the order of keys. Is that what you wanted?

Reply:
Arg, good point. We probably have to stringify both sides just to get ordered object equalty checks.

Reply:
Probably the best way. We could also cycle through `Object.entries` recursively...?

Reply:
Okay I think those test revealed that an object sorting API is not the most clever thing to have. I changed the interface a bit to combine sording and serialization which is what we really want: sorted JSON serialization.